services/wallet/market/market.go

```diff
@@ -46,7 +59,8 @@ func NewManager(providers []thirdparty.MarketDataProvider, feed *event.Feed) *Ma
 
 	return &Manager{
 		feed:           feed,
-		priceCache:     make(DataPerTokenAndCurrency),
+		priceCache:     *NewCache(make(TokenPriceCache)),
+		marketCache:    *NewCache(make(TokenMarketCache)),
```

Comment:
These are the only 2 places where `NewCache` is used. And in both lines we convert it from pointer to value: `*NewCache`. Probably makes sense to either keep `priceCache` and `marketCache` as pointers, or just return value from `NewCache`.

Actually, why are `priceCache`/`marketCache` stored not by pointers? 🤔

Reply:
Yeah good questions, I’m not sure about what’s best here, but I’ll share some thoughts.

I think `NewCache` should return a pointer because it’s constructing a new instance, and if we return the pointer I think we can avoid an extra memory copy (not sure about this, but I think that’s true)

We could store the instances of `MarketCache` as pointers, but I don’t think we need to because we’re owning the instance inside the `Manager` struct and we use a pointer to that struct in the other places. Though if we store them as pointers we could have the benefit of sharing them between `Manager` structs